### PR TITLE
Add support for email validation to hvclient

### PR DESCRIPTION
**Pull request by toddgaunt**
# HVClient PR Testing
hvclient.conf file:
```json
{
	"url": "http://localhost:3011",
	"api_key": "7ffbbf9281266c93",
	"api_secret": "701ea4b3a74db3d2a655dc27386828edd56cc58c",
	"cert_file": "",
	"key_file": "",
	"key_passphrase": "",
	"insecure_skip_verify": false,
	"timeout": 60
}
```
Submit a claim:
```
toddgaunt@MacBook-Pro[development:~/go/src/globalsign/hvpki/hvca/local]% hvclient -claimsubmit=atlasqa.co.uk -config ./hvclient.conf
globalsign-domain-verification=3655054A92FB74ACE31C396BFDB1CC90,2022-05-06 19:06:19 +0000 UTC,0B9CD2D782D67226B1DAA2CFEF62A154
```
Retrieve a claim:
```
toddgaunt@MacBook-Pro[development:~/go/src/globalsign/hvpki/hvca/local]% hvclient -claimretrieve=0B9CD2D782D67226B1DAA2CFEF62A154 -config ./hvclient.conf
0B9CD2D782D67226B1DAA2CFEF62A154,PENDING,atlasqa.co.uk.,2022-04-06 19:06:19 +0000 UTC,2022-05-06 19:06:19 +0000 UTC
```
Verify a claim via email:
```
toddgaunt@MacBook-Pro[development:~/go/src/globalsign/hvpki/hvca/local]% hvclient -claimemail=0B9CD2D782D67226B1DAA2CFEF62A154 -address=admin@atlasqa.co.uk  -config ./hvclient.conf
CREATED
```
Retrieve the claim (still pending because I don't know how to verify the domain via the email sent to admin@atlasqa.co.uk):
```
toddgaunt@MacBook-Pro[development:~/go/src/globalsign/hvpki/hvca/local]% hvclient -claimretrieve=0B9CD2D782D67226B1DAA2CFEF62A154 -config ./hvclient.conf
0B9CD2D782D67226B1DAA2CFEF62A154,PENDING,atlasqa.co.uk.,2022-04-06 19:06:19 +0000 UTC,2022-05-06 19:06:19 +0000 UTC
```
Retrieve the claim email list:
```
toddgaunt@MacBook-Pro[development:~/go/src/globalsign/hvpki/hvca/local]% hvclient -claimemaillist=0B9CD2D782D67226B1DAA2CFEF62A154 -config ./hvclient.conf
Constructed: [admin@atlasqa.co.uk administrator@atlasqa.co.uk webmaster@atlasqa.co.uk hostmaster@atlasqa.co.uk postmaster@atlasqa.co.uk]
DNS: [hostmaster@gandi.net]
Errors:
```

### CODE REVIEW

File: client_api.go [hunk 1]
```diff
@@ -413,7 +438,7 @@ func (c *Client) ClaimDNS(ctx context.Context, id, authDomain string) (bool, err
 	return c.claimAssert(ctx, body, id, pathDNS)
 }
 
-// ClaimHTTP requests assertion of domain control using HTTP once the appropriate
+// ClaimHTTP requests assEmailertion of domain control using HTTP once the appropriate
```
Comment (mara-soldan) [inline client_api.go:441]: why was this changed ?

File: mock_hvca_test.go [hunk 2]
```diff
@@ -594,9 +667,11 @@ func mockTrustChain(w http.ResponseWriter, r *http.Request) {
 
 // mockNotImplemented is a stub handler that writes a 501 not implemented
 // response.
+/* This is commented out because the linter is complaining >:(
 func mockNotImplemented(w http.ResponseWriter, r *http.Request) {
```
Comment (mara-soldan) [inline mock_hvca_test.go:671]: is this still used ? if not we cannot remove it because the linter is complaining?

File: client_api.go [hunk 3]
```diff
@@ -426,6 +451,41 @@ func (c *Client) ClaimHTTP(ctx context.Context, id, authDomain, scheme string) (
 	return c.claimAssert(ctx, body, id, pathHTTP)
 }
 
+// ClaimEmail requests assertion of domain control using Email once the appropriate
```
Comment (robplee) [inline client_api.go:454]: This isn't how email domain claims work.  Posting to /email with an email address in the body requests that a link be sent to that email address which will be followed by the user in order to prove ownership of the domain

File: client_mock_test.go [hunk 4]
```diff
@@ -544,6 +544,106 @@ func TestClientMockClaimHTTP(t *testing.T) {
 	}
 }
 
+func TestClientMockClaimEmail(t *testing.T) {
```
Comment (robplee) [inline client_mock_test.go:547]: Not sure of the hvclient conventions but normally this would be named something like:
TestClient_ClaimEmail

File: mock_hvca_test.go (hunk 2 above)
Comment (robplee) [reply mock_hvca_test.go:671]: Alternatively: what's the complain the linter is making?  Maybe it can be resolved if this code is useful

Comment (toddgaunt) [reply mock_hvca_test.go:671]: This was used to mock unimplemented functions. I'd rather  it just stick around since  its one of those puzzle pieces thats nice to have lying around. Otherwise we remove it, and then when we need to mock another endpoint we rewrite it.

File: client_api.go (hunk 3 above)
Comment (toddgaunt) [reply client_api.go:454]: Yes I think I just neglected the comment clean up, good spot

File: client_mock_test.go (hunk 4 above)
Comment (toddgaunt) [reply client_mock_test.go:547]: No underscores are used in hvclient like that so far, so I was sticking to existing conventions in codebase here.

File: client_api.go (hunk 1 above)
Comment (toddgaunt) [reply client_api.go:441]: My mistake that I didn't notice, thanks!

File: client_api.go [hunk 5]
```diff
@@ -426,6 +451,42 @@ func (c *Client) ClaimHTTP(ctx context.Context, id, authDomain, scheme string) (
 	return c.claimAssert(ctx, body, id, pathHTTP)
 }
 
+// ClaimEmail requests for an email with a verification link be sent to the
+// provided emailAddress in order for the user to assert control of domain
+// control by following the link inside the sent email. A return value of false
```
Comment (angela-globalsign) [inline client_api.go:456]:
```suggestion
// by following the link inside the sent email. A return value of false
```
One control is extra no?
.... to assert control of domain control by following....

File: mock_hvca_test.go (hunk 2 above)
Comment (toddgaunt) [reply mock_hvca_test.go:671]: To answer the question, the linter is complaining about an unused function.

**Review by angela-globalsign — APPROVED** [review]

File: mock_hvca_test.go (hunk 2 above)
Comment (mara-soldan) [reply mock_hvca_test.go:671]: I think better to remove it, ideally we would not mock any endpoint in the future like this

**Review by mara-soldan — APPROVED** [review]

**Review by angela-globalsign — APPROVED** [review]

File: client_mock_test.go (hunk 4 above)
Comment (robplee) [reply client_mock_test.go:547]: Fine by me.

**Review by robplee — APPROVED** [review]
